sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -0,0 +1,106 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.android.core.internal.util.MainThreadChecker;
+import io.sentry.util.Objects;
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+// import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public final class ScreenshotEventProcessor
+    implements EventProcessor, Application.ActivityLifecycleCallbacks, Closeable {
+
+  private final @NotNull Application application;
+  private final @NotNull SentryAndroidOptions options;
+  //  TODO: private final @NotNull WeakReference<Activity> currentActivity = new WeakReference();
+  private @Nullable Activity activity;
+
+  public ScreenshotEventProcessor(
+      final @NotNull Application application, final @NotNull SentryAndroidOptions options) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+
+    if (this.options.isAttachScreenshot()) {
+      application.registerActivityLifecycleCallbacks(this);
+    }
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public @NotNull SentryEvent process(
+      final @NotNull SentryEvent event, @Nullable Map<String, Object> hint) {
+    if (options.isAttachScreenshot()
+        && event.isErrored()
+        && MainThreadChecker.isMainThread(Thread.currentThread())) {
+      if (activity != null
+          && activity.getWindow() != null
+          && activity.getWindow().getDecorView() != null) {
+        View view = activity.getWindow().getDecorView().getRootView();
+        // TODO: check if using Canvas is better
+        final boolean cacheEnabled = view.isDrawingCacheEnabled();
+        view.setDrawingCacheEnabled(true);
+        view.buildDrawingCache(true);
+
+        Bitmap bitmap = Bitmap.createBitmap(view.getDrawingCache());
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        bitmap.compress(Bitmap.CompressFormat.PNG, 0, byteArrayOutputStream);
+
+        if (hint == null) {
+          hint = new HashMap<>();
+        }
```

Comment:
This required a workaround since objects are passed by reference, creating `hint` here won't help.

Reply:
Since `hint` is most of the time null, the SDK would create a new instance before processing the event, so event processors, integrations, etc, can attach/remove data to the hint.
@bruno-garcia opinions?

Reply:
good point, do the other SDKs always pass an instance of `Hint`? e.g: Python?
I believe it might make sense to always pass an instance through beforeSend etc by default

Reply:
Python creates an instance if not yet https://github.com/getsentry/sentry-python/blob/4703bc35a9a5d65d6187ad1b0838a201e1c6e25d/sentry_sdk/client.py#L329